FormalConjectures/ErdosProblems/728.lean

```diff
@@ -22,22 +22,30 @@ import FormalConjectures.Util.ProblemImports
 *Reference:* [erdosproblems.com/728](https://www.erdosproblems.com/728)
 -/
 
+open Real
+open scoped Nat
+
 namespace Erdos728
 
 /--
-Let $\varepsilon, C > 0$. Are there integers $a, b, n$ such that
-$$a > \varepsilon n,\quad b > \varepsilon n, \quad a!\, b! \mid n!\, (a + b - n)!, $$
+Let $\varepsilon, C, C' > 0$. Are there integers $a, b, n$ such that
+$$\varepsilon n < a < (1 - \varepsilon) n,\quad \varepsilon n < b < (1 - \varepsilon) n,
+  \quad a!\, b! \mid n!\, (a + b - n)!, $$
 and
-$$ a + b > n + C \log n ?$$
+$$C \log n < a + b - n < C' \log n ?$$
 -/
 @[category research open, AMS 11]
-theorem erdos_728 :
-    (∀ (ε C : ℝ) (hε : 0 < ε) (hC : 0 < C), ∃ a b n : ℕ,
+theorem erdos_728 : answer(sorry) ↔
+    ∀ (ε C C' : ℝ) (hε : 0 < ε) (hC : 0 < C) (hCC' : C < C'),
+      ∃ a b n : ℕ,
       0 < n ∧
       ε * n < a ∧
       ε * n < b ∧
-      Nat.factorial a * Nat.factorial b ∣ Nat.factorial n * Nat.factorial (a + b - n) ∧
-      a + b > n + C * Real.log n) ↔ answer(sorry) := by
+      a < (1 - ε) * n ∧
+      b < (1 - ε) * n ∧
+      a ! * b ! ∣ n ! * (a + b - n)! ∧
+      a + b > n + C * log n ∧
+      a + b < n + C' * log n := by
   sorry
```

Comment:
When putting `answer(False)` above, then AlphaProof finds
```
  use default, (by match· _ _ _ one_pos one_pos one_lt_two with | ⟨e,y,k,s, a, right, M, _⟩ =>linarith only[a.trans M])
```
I think this corresponds to the counter example with `ε = C = 1` and `C' = 2`.